pkg/azurefile/azure.go

```diff
@@ -102,3 +107,63 @@ func getKubeClient(kubeconfig string) (*kubernetes.Clientset, error) {
 
 	return kubernetes.NewForConfig(config)
 }
+
+func updateSubnetServiceEndpoints(ctx context.Context, az *azure.Cloud, subnetLockMap *lockMap) error {
+	if az == nil {
+		return fmt.Errorf("the cloud parameter is nil")
+	}
+	if subnetLockMap == nil {
+		return fmt.Errorf("the subnet lockMap parameter is nil")
+	}
+
+	resourceGroup := az.ResourceGroup
+	location := az.Location
+	vnetName := az.VnetName
+	subnetName := az.SubnetName
+
+	subnet, err := az.SubnetsClient.Get(ctx, resourceGroup, vnetName, subnetName, "")
+	if err != nil {
+		return fmt.Errorf("failed to get the subnet %s under vnet %s: %v", subnetName, vnetName, err)
+	}
+	endpointLocaions := []string{location}
+	storageServiceEndpoint := network.ServiceEndpointPropertiesFormat{
+		Service:   &storageService,
+		Locations: &endpointLocaions,
+	}
+	storageServiceExists := false
+	if subnet.SubnetPropertiesFormat != nil {
+		if subnet.SubnetPropertiesFormat.ServiceEndpoints != nil {
```

Comment:
```
if subnet.SubnetPropertiesFormat.ServiceEndpoints == nil {
   subnet.SubnetPropertiesFormat.ServiceEndpoints = &[]network.ServiceEndpointPropertiesFormat{storageServiceEndpoint}
}
```

by this way, could reduce one level of if-else

Reply:
done